classes/TaskRunner/Upgrade/UninstallIncompatibleModules.php

```diff
@@ -0,0 +1,158 @@
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ */
+
+namespace PrestaShop\Module\AutoUpgrade\TaskRunner\Upgrade;
+
+use PrestaShop\Module\AutoUpgrade\TaskRunner\AbstractTask;
+use PrestaShop\Module\AutoUpgrade\UpgradeContainer;
+use PrestaShop\Module\AutoUpgrade\UpgradeException;
+use PrestaShop\Module\AutoUpgrade\UpgradeTools\SettingsFileWriter;
+
+/**
+ * Uninstall modules that are not compatible with the version of PrestaShop being installed
+ */
+class UninstallIncompatibleModules extends AbstractTask
+{
+    // Modules incompatible with 1.7.x
+    private static $incompatibleModules = [
+        'bankwire',
+        'blockbanner',
+        'blockcart',
+        'blockcategories',
+        'blockcms',
+        'blockcmsinfo',
+        'blockcontact',
+        'blockcurrencies',
+        'blocklanguages',
+        'blocklayered',
+        'blockmyaccount',
+        'blocknewsletter',
+        'blocksearch',
+        'blocksocial',
+        'blocktopmenu',
+        'blockuserinfo',
+        'cheque',
+        'homefeatured',
+        'homeslider',
+        'onboarding',
+        'socialsharing',
+        'vatnumber',
+        'blockadvertising',
+        'blockbestsellers',
+        'blockcustomerprivacy',
+        'blocklink',
+        'blockmanufacturer',
+        'blocknewproducts',
+        'blockpermanentlinks',
+        'blockrss',
+        'blocksharefb',
+        'blockspecials',
+        'blocksupplier',
+        'blockviewed',
+        'crossselling',
+        'followup',
+        'productscategory',
+        'producttooltip',
+        'mailalert',
+        'blockcontactinfos',
+        'blockfacebook',
+        'blockmyaccountfooter',
+        'blockpaymentlogo',
+        'blockstore',
+        'blocktags',
+        'blockwishlist',
+        'productpaymentlogos',
+        'sendtoafriend',
+        'themeconfigurator',
+    ];
+
+    public function run()
+    {
+        $this->next = 'upgradeDb';
+        if (version_compare($this->container->getState()->getInstallVersion(), '1.7.0.0', '<')) {
+            $this->logger->info(
+                $this->translator->trans(
+                    'No incompatible module to uninstall. Now upgrading database...',
+                    array(),
+                    'Modules.Autoupgrade.Admin'
+                )
+            );
+
+            return true;
+        }
+        $modulePath = $this->container->getProperty(UpgradeContainer::PS_ROOT_PATH)
+            . DIRECTORY_SEPARATOR . 'modules' . DIRECTORY_SEPARATOR;
+
+        if (!is_dir($modulePath)) {
```

Comment:
Suggestion: put this into private method `$this->validateModulePath()` to make the reading of `run()` easier